### ISSUE TITLE
Potential error on order edit page when address has extension attributes

### ISSUE THREAD

**Issue by dankocherga** [magento/magento2#10438]
Potential error on order edit page when address has extension attributes

### Preconditions
1. Magento 2.1.7
2. PHP 7.0.18

### Steps to reproduce
1. Suppose you init extension attributes of the order address on the `sales_order_address_load_after` event (any other way is also ok): 
```php
        $address = $observer->getEvent()->getAddress();
        $extensionAttributes = $address->getExtensionAttributes();
        if (!$extensionAttributes) {
            $extensionAttributes = $this->extensionAttributesFactory->create();
            $address->setExtensionAttributes($extensionAttributes);
        }
```
2. Go to order's admin panel and click "Edit"

### Expected result
1. Order edit page opens

### Actual result
1. Order edit page fails with the following error: `Recoverable Error: Object of class Magento\Sales\Api\Data\OrderAddressExtension could not be converted to string in .../module-sales/Model/AdminOrder/Create.php on line 503`

The code on this line does the following:
```php
        $shippingAddress = $order->getShippingAddress();
        if ($shippingAddress) {
            $addressDiff = array_diff_assoc($shippingAddress->getData(), $order->getBillingAddress()->getData());
            unset($addressDiff['address_type'], $addressDiff['entity_id']);
            $shippingAddress->setSameAsBilling(empty($addressDiff));
        }
```

The problem is that `$shippingAddress` object could potentially have `extension_attributes` field in the data and that field is an object (which is correct).
When you diff arrays with object in a field you get the error:

```cli
php > var_dump(array_diff_assoc(['foo' => new stdClass], ['foo' => new stdClass]));
PHP Catchable fatal error:  Object of class stdClass could not be converted to string in php shell code on line 1
``` 

`array_diff_assoc` converts values to strings according to the [documentation](http://php.net/manual/en/function.array-diff-assoc.php):

> Two values from key => value pairs are considered equal only if (string) $elem1 === (string) $elem2 . In other words a strict check takes place so the string representations must be the same.

**Comment by magento-engcom-team** [comment]
@dankocherga From github code I can only get error that missing argument in $this->extensionAttributesFactory->create() function. When I tried to customize code I would not receive any error. Pls show all your module code.

**Comment by dankocherga** [comment]
@magento-engcom-team 
Please look at this sample module: https://github.com/dankocherga/m2-issue-10438
It adds extension attributes to order's shipping and billing address.
Try creating an order and editing it via admin panel.

**Comment by magento-engcom-team** [comment]
@dankocherga, thank you for your report.
We've created internal ticket(s) MAGETWO-75789 to track progress on the issue.

**Comment by YevSent** [comment]
I'm working on it #SQUASHTOBERFEST

**Comment by okorshenko** [comment]
thank you for your report. The issue has been fixed and delivered to 2.2-develop branch. Will be available with upcoming patch release

**Comment by magento-engcom-team** [comment]
Hi @dankocherga. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1362 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - 4705e23479eae053947c52ee5be6969a79a82079
 - 6d355e033e513cefc390f26417eb2321756b1dce
 - 77f34cf7e3e12f567aa8fdd6918bcf9cd1ffda2f
 - 748ab4db199379ce2154ffb63b9801a8c0d13bb4
 - 90f9d95056b1cd7fe0be036ec72c236bc03738fc
 - 44ca24fdc0d3367515893cf9a8fc1bffcc070140
 - f0b7f2b1708218a8d894178fc9a4f045dfdbdedd
 - c3e1d93db404b63dc38050f9089d2b6a72094085
 - 4810755a4751e3a389e2a39dcbaf55ab72c2149c
 - 4709a222e58918876e2f5b9d28b8f0e5e120d52f
 - a2b9fab4b10421c327b1f323bf222750ac371fb9
 - ac7df33cecf744cf9899067dc62453bc7a15ca50
 - 245238dcbb72b10fad84596399b12b9814295641
 - 2c153c3c428308c1cbc1969bd2eb271913d6551d
 - 48a72fa0d214418a79a5d77d25ead1eb75cbc1c1
 - 625f685c183e0c0a7993ca38c025a079e0e29b47
 - afa4516615808a427be8a160e9cd6ae80c321b27

The fix will be available with the upcoming 2.3.0 release.
